modules/secure-cloud-run-core/loadbalancer.tf

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module "lb-http" {
+  source                          = "GoogleCloudPlatform/lb-http/google//modules/serverless_negs"
+  version                         = "~> 5.1"
+  name                            = var.lb_name
+  project                         = var.serverless_project_id
+  ssl                             = var.ssl
+  managed_ssl_certificate_domains = [var.domain]
+  https_redirect                  = var.ssl
+
+  backends = {
+    default = {
+      description = null
+      groups = [
+        {
+          group = google_compute_region_network_endpoint_group.serverless_neg.id
+        }
+      ]
+      enable_cdn              = false
+      security_policy         = google_compute_security_policy.cloud-armor-security-policy.id
+      custom_request_headers  = null
+      custom_response_headers = null
+
+      iap_config = {
+        enable               = false
+        oauth2_client_id     = ""
+        oauth2_client_secret = ""
+      }
+      log_config = {
+        enable      = false
+        sample_rate = null
+      }
+    }
+  }
+}
+
+resource "google_compute_region_network_endpoint_group" "serverless_neg" {
+  provider              = google-beta
+  name                  = "serverless-neg"
+  project               = var.serverless_project_id
+  network_endpoint_type = "SERVERLESS"
+  region                = var.region
+  cloud_run {
+    service = var.service_name
+  }
+}
+
+resource "google_compute_security_policy" "cloud-armor-security-policy" {
+  project = var.serverless_project_id
+  name    = "cloud-armor-waf-policy"
+  dynamic "rule" {
+    for_each = var.default_rules
+    content {
+      action      = rule.value.action
+      priority    = rule.value.priority
+      description = rule.value.description
+      match {
+        versioned_expr = rule.value.versioned_expr
+        config {
+          src_ip_ranges = rule.value.src_ip_ranges
+        }
+      }
+    }
+  }
+  dynamic "rule" {
+    for_each = var.owasp_rules
+    content {
+      action   = rule.value.action
+      priority = rule.value.priority
+      match {
+        expr {
+          expression = rule.value.expression
+        }
+      }
+    }
```

Comment:
We can merge maps in a local and use a single dynamic which is cleaner

Reply:
We should open a new issue similar to https://github.com/GoogleCloudPlatform/terraform-google-cloud-run/issues/46 to address this. The new issue cannot to be opened right now since we don't have Cloud Armor in the code. It will be address in the secure-cloud-run submodule.